docs/source/en/quantization/bitsandbytes.md

```diff
@@ -13,88 +13,240 @@ specific language governing permissions and limitations under the License.
+4-bit quantization compresses a model even further, and it is commonly used with
+[QLoRA](https://hf.co/papers/2305.14314) to finetune quantized LLMs.
 
+We'll work with the
+[FLUX.1-dev model](https://huggingface.co/black-forest-labs/FLUX.1-dev),
+demonstrating how quantization can help you run it on less than 16GB of VRAM—even on a free Google
+Colab instance.
+
+![comparison image](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/quant-bnb/comparison.png)
 
 To use bitsandbytes, make sure you have the following libraries installed:
 
 ```bash
 pip install diffusers transformers accelerate bitsandbytes -U
 ```
 
-Now you can quantize a model by passing a [`BitsAndBytesConfig`] to [`~ModelMixin.from_pretrained`]. This works for any model in any modality, as long as it supports loading with [Accelerate](https://hf.co/docs/accelerate/index) and contains `torch.nn.Linear` layers.
+Now you can quantize a model by passing a [`BitsAndBytesConfig`] to [`~ModelMixin.from_pretrained`].
+This works for any model in any modality, as long as it supports loading with
+[Accelerate](https://hf.co/docs/accelerate/index) and contains `torch.nn.Linear` layers.
 
 <hfoptions id="bnb">
 <hfoption id="8-bit">
 
 Quantizing a model in 8-bit halves the memory-usage:
 
+As `bitsandbytes` is supported in both `transformers` and `diffusers` we can quantize both the
+`FluxTransformer2DModel` and `T5EncoderModel`.
+
 ```py
-from diffusers import FluxTransformer2DModel, BitsAndBytesConfig
+from diffusers import BitsAndBytesConfig as DiffusersBitsAndBytesConfig
+from transformers import BitsAndBytesConfig as TransformersBitsAndBytesConfig
 
-quantization_config = BitsAndBytesConfig(load_in_8bit=True)
+from diffusers import FluxTransformer2DModel
+from transformers import T5EncoderModel
 
-model_8bit = FluxTransformer2DModel.from_pretrained(
-    "black-forest-labs/FLUX.1-dev", 
+quant_config = TransformersBitsAndBytesConfig(
+    load_in_8bit=True,
+)
+
+text_encoder_2_8bit = T5EncoderModel.from_pretrained(
+    "black-forest-labs/FLUX.1-dev,
+    subfolder="text_encoder_2",
+    quantization_config=quant_config,
+    torch_dtype=torch.float16,
+)
+
+quant_config = DiffusersBitsAndBytesConfig(
+    load_in_8bit=True,
+)
+
+transformer_8bit = FluxTransformer2DModel.from_pretrained(
+    "black-forest-labs/FLUX.1-dev",
     subfolder="transformer",
-    quantization_config=quantization_config
+    quantization_config=quant_config,
+    torch_dtype=torch.float16,
 )
 ```
 
-By default, all the other modules such as `torch.nn.LayerNorm` are converted to `torch.float16`. You can change the data type of these modules with the `torch_dtype` parameter if you want:
+By default, all the other modules such as `torch.nn.LayerNorm` are converted to `torch.float16`.
+You can change the data type of these modules with the `torch_dtype` parameter if you want:
 
 ```py
-from diffusers import FluxTransformer2DModel, BitsAndBytesConfig
```

Comment:
I don't think you need to show the entire code snippet here since it is a bit lengthy and redundant. You can just do something like the below:

```py
text_encoder_2_8bit = T5EncoderModel.from_pretrained(
    "black-forest-labs/FLUX.1-dev,
    subfolder="text_encoder_2",
    quantization_config=quant_config,
    torch_dtype=torch.float32,
)
```